libfwupdplugin/fu-common.c

```diff
@@ -219,34 +219,58 @@ fu_common_align_up(gsize value, guint8 alignment)
 }
 
 /**
- * fu_battery_state_to_string:
- * @battery_state: a battery state, e.g. %FU_BATTERY_STATE_FULLY_CHARGED
+ * fu_power_state_to_string:
+ * @power_state: a power state, e.g. %FU_POWER_STATE_AC_FULLY_CHARGED
  *
  * Converts an enumerated type to a string.
  *
  * Returns: a string, or %NULL for invalid
  *
- * Since: 1.6.0
+ * Since: 1.8.11
  **/
 const gchar *
-fu_battery_state_to_string(FuBatteryState battery_state)
+fu_power_state_to_string(FuPowerState power_state)
 {
-	if (battery_state == FU_BATTERY_STATE_UNKNOWN)
+	if (power_state == FU_POWER_STATE_UNKNOWN)
 		return "unknown";
-	if (battery_state == FU_BATTERY_STATE_CHARGING)
-		return "charging";
-	if (battery_state == FU_BATTERY_STATE_DISCHARGING)
-		return "discharging";
-	if (battery_state == FU_BATTERY_STATE_EMPTY)
-		return "empty";
-	if (battery_state == FU_BATTERY_STATE_FULLY_CHARGED)
-		return "fully-charged";
+	if (power_state == FU_POWER_STATE_BATTERY)
+		return "battery";
+	if (power_state == FU_POWER_STATE_BATTERY_DISCHARGING)
+		return "battery-discharging";
+	if (power_state == FU_POWER_STATE_BATTERY_EMPTY)
+		return "battery-empty";
+	if (power_state == FU_POWER_STATE_AC)
+		return "ac";
+	if (power_state == FU_POWER_STATE_AC_CHARGING)
+		return "ac-charging";
+	if (power_state == FU_POWER_STATE_AC_FULLY_CHARGED)
+		return "ac-fully-charged";
```

Comment:
technically I think you're missing a state here of `ac-discharging` when you have a power adapter plugged in but it's small and battery is going down

Reply:
Yup, I'm struggling with this one; if the AC adapter is plugged in, but the battery is actually _going down_ is that conceptually "on AC power"? I'd argue it's **relying** on the battery being present, and no longer on AC power.

Reply:
you would see a little AC adapter plug in your DE and everywhere else in the kernel would report it to be plugged in.  So I'd say `ac-discharging`

Reply:
...but would you expect it to allow firmware updates to start? I'm guessing the UEFI updater is only really checking the AC input ACPI input.

Reply:
> I'm guessing the UEFI updater is only really checking the AC input ACPI input.

Yeah exactly.

Reply:
So in the AC plugged in case, but battery actually slowly discharging case - I'd argue you should set just FU_POWER_STATE_AC.